conan/internal/util/files.py

```diff
@@ -256,15 +261,46 @@ def mkdir(path):
     os.makedirs(path)
 
 
-def tar_extract(fileobj, destination_dir):
+def tar_extract(fileobj, destination_dir, compression_plugin=None, conf=None):
+    if compression_plugin:
+        _tar_extract_with_plugin(fileobj, destination_dir, compression_plugin, conf)
+        return
+
     the_tar = tarfile.open(fileobj=fileobj)
     # NOTE: The errorlevel=2 has been removed because it was failing in Win10, it didn't allow to
     # "could not change modification time", with time=0
     # the_tar.errorlevel = 2  # raise exception if any error
     the_tar.extraction_filter = (lambda member, path: member)  # fully_trusted, avoid Py3.14 break
     the_tar.extractall(path=destination_dir)
     the_tar.close()
-
+    if list(Path(destination_dir).glob(f"{COMPRESSED_PLUGIN_TAR_NAME}.*")):
+        raise ConanException(f"Error while extracting {os.path.basename(fileobj.name)}.\n"
+                             "This file has been compressed using a `compression` plugin.\n"
+                             "If your organization uses this plugin, ensure it is correctly installed on your environment.")
+
+def _tar_extract_with_plugin(fileobj, destination_dir, compression_plugin, conf):
+    """First remove tar.gz wrapper and then call the plugin to extract"""
+    with tempfile.TemporaryDirectory() as temp_dir:
+        t1 = time.time()
+        the_tar = tarfile.open(fileobj=fileobj)
+        the_tar.extraction_filter = (lambda member, path: member)  # fully_trusted, avoid Py3.14 break
+        the_tar.extractall(path=temp_dir)
+        extracted_file = the_tar.getnames()[0]
+        the_tar.close()
+        # Check if the tar was compressed with the compression plugin by checking the existence of
+        # our constant COMPRESSED_PLUGIN_TAR_NAME (without extension as extension is added by the plugin)
+        if list(Path(temp_dir).glob(f"{COMPRESSED_PLUGIN_TAR_NAME}.*")):
+            # Get the only extracted file: the plugin tar
+            plugin_tar_path = os.path.join(temp_dir, extracted_file)
+            ConanOutput().debug(f"Unwrapped in {time.time() - t1} time")
+            t1 = time.time()
+            compression_plugin.tar_extract(archive_path=plugin_tar_path, dest_dir=destination_dir, conf=conf)
+            ConanOutput().debug(f"Extracted in {time.time() - t1} time on plugin")
+        else:
+            # The tar was not compressed using the plugin, copy files to destination
+            from conan.tools.files import copy
+            ConanOutput().debug(f"Extracted in {time.time() - t1} time built in")
+            copy(None, pattern="*", src=temp_dir, dst=destination_dir)
```

Comment:
The contents will already be unpacked here and will just be deleted on cleanup of `temp_dir`. Instead of copying, would you consider moving them directly? That would seem to improve I/O required, skip the double disk space usage requirement, and likely result in much less I/O overhead from antivirus scanning on Windows.